www/js/metrics/MetricsTab.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState, useMemo } from "react";
+import { angularize, getAngularService } from "../angular-react-helper";
+import { View, ScrollView, useWindowDimensions } from "react-native";
+import { Appbar } from "react-native-paper";
+import NavBarButton from "../components/NavBarButton";
+import { useTranslation } from "react-i18next";
+import { DateTime } from "luxon";
+import { MetricsData } from "./metricsTypes";
+import MetricsCard from "./MetricsCard";
+import { formatForDisplay, useImperialConfig } from "../config/useImperialConfig";
+import MetricsDateSelect from "./MetricsDateSelect";
+import WeeklyActiveMinutesCard from "./WeeklyActiveMinutesCard";
+import { secondsToHours, secondsToMinutes } from "./metricsHelper";
+import CarbonFootprintCard from "./CarbonFootprintCard";
+import Carousel from "../components/Carousel";
+import DailyActiveMinutesCard from "./DailyActiveMinutesCard";
+import CarbonTextCard from "./CarbonTextCard";
+import ActiveMinutesTableCard from "./ActiveMinutesTableCard";
+
+export const METRIC_LIST = ['duration', 'mean_speed', 'count', 'distance'] as const;
+
+async function fetchMetricsFromServer(type: 'user'|'aggregate', dateRange: DateTime[]) {
+  const CommHelper = getAngularService('CommHelper');
+  const query = {
+    freq: 'D',
+    start_time: dateRange[0].toSeconds(),
+    end_time: dateRange[1].toSeconds(),
+    metric_list: METRIC_LIST,
+    is_return_aggregate: (type == 'aggregate'),
+  }
+  if (type == 'user')
+    return CommHelper.getMetrics('timestamp', query);
+  return CommHelper.getAggregateData("result/metrics/timestamp", query);
+}
+
+function getLastTwoWeeksDtRange() {
+  const now = DateTime.now().startOf('day');
+  const start = now.minus({ days: 15 });
+  const end = now.minus({ days: 1 });
```

Comment:
for the record, we might want to revisit this later if we ever change our labeling flow to be more real-time, so that the labels are reflected in the metrics. Note also that this is in UTC, which may cause some confusion among users that we could consider addressing in a future cleanup.